src/diffusers/models/unet_2d_condition.py

```diff
@@ -230,16 +230,16 @@ def forward(
         # 1. time
         timesteps = timestep
         if not torch.is_tensor(timesteps):
+            # TODO: this requires sync between CPU and GPU. So try to pass timesteps as tensors if you can
             timesteps = torch.tensor([timesteps], dtype=torch.long, device=sample.device)
```

Comment:
Do we need int64 tensors for timesteps, no matter the pipeline's precision?

Reply:
At least for all the Stable Diffusion applications I've seen so far, timesteps are ints in the range 0..1000.

Even if other diffusion models do several orders of magnitude more than that, you'd think `torch.int` would be plenty.

Unless there's some byte alignment optimization reason to specifically make them 64-bit?